server/publish.js

```diff
@@ -1,70 +1,85 @@
 Meteor.publish("chatterMessages", function (opts) {
-    check(opts, {
-        roomId: String,
-        messageLimit: Number
-    });
+  check(opts, {
+    roomId: String,
+    messageLimit: Number
+  });
 
-    /*
-    let room = ChatterRoom.findOne({ name: opts.roomName });
-    if (_.isUndefined(room)) return [];
-    // TODO Check for room restrictions here
-    */
+  /*
+  let room = ChatterRoom.findOne({ name: opts.roomName });
+  if (_.isUndefined(room)) return [];
+  // TODO Check for room restrictions here
+  */
 
-    return ChatterMessage.find({
-        roomId: opts.roomId,
-    }, {
-        limit: opts.limit,
-        fields: {
-            message: 1,
-            roomId: 1,
-            userNick: 1,
-            userAvatar: 1,
-            userId: 1,
-            createdAt: 1
-        }
-    });
+  return ChatterMessage.find({
+    roomId: opts.roomId,
+  }, {
+    limit: opts.limit,
+    fields: {
+      message: 1,
+      roomId: 1,
+      userNick: 1,
+      userAvatar: 1,
+      userId: 1,
+      createdAt: 1
+    }
+  });
 });
 
 Meteor.publish("chatterRooms", function () {
-    let user = Meteor.users.findOne(this.userId);
+  let user = Meteor.users.findOne(this.userId);
 
-    // TODO do some clever things here to determine
-    // which rooms are accessible
+  // TODO do some clever things here to determine
+  // which rooms are accessible
 
-    return ChatterRoom.find({
-    }, {
-        fields: {
-            name: 1,
-            roomType: 1,
-            lastActive: 1
-        }
-    });
+  return ChatterRoom.find({
+  }, {
+    fields: {
+      name: 1,
+      description: 1,
+      roomType: 1,
+      lastActive: 1,
+      archived: 1,
+      createdAt: 1
+    }
+  });
 });
 
 Meteor.publish("chatterUserRooms", function () {
-    let user = Meteor.users.findOne(this.userId);
+  let user = Meteor.users.findOne(this.userId);
 
-    // TODO do some clever things here to determine
-    // which rooms are accessible
+  // TODO do some clever things here to determine
+  // which rooms are accessible
 
-    return ChatterUserRoom.find({
-    }, {
-        fields: {
-            count: 1,
-            userId: 1,
-            roomId: 1
-        }
-    });
+  return ChatterUserRoom.find({
+  }, {
+    fields: {
+      count: 1,
+      userId: 1,
+      roomId: 1
+    }
+  });
 });
 
-Meteor.publish("chatterUsers", function (roomId) {
-    // TODO do some clever things here to determine
-    // which users are visible
-    return Meteor.users.find({
-        "status.online": true
-    }, {
-        userNick: 1,
-        avatar: 1
-    });
+Meteor.publish("users", function (roomId) {
+  // TODO do some clever things here to determine
+  // which users are visible
+  return Meteor.users.find({
+    "status.online": true
+  }, {
+    userNick: 1,
```

Comment:
Added a roomLimit key to Chatter.options. Couple of questions regarding this practice though. 
Right now the limit is passed in through the client call as in:

 Meteor.call("room.create", {messageLimit: Chatter.options.messageLimit});

Is there any reason why this cant be simply set in the server directly?

The other question might be stupid, but what happens to the messages that do not get sent? Is there some sort of pagination that needs to be implemented?